tests/test_simple_types.py

```diff
@@ -49,7 +49,7 @@ def __str__(self):
     (IonPyInt, e_int(2 ** 64 + 1)),
     (IonPyFloat, e_float(1e0)),
     (IonPyDecimal, e_decimal(Decimal('1.1'))),
-    (IonPyTimestamp, e_timestamp(datetime(2012, 1, 1))),
+    (IonPyTimestamp, e_timestamp(Timestamp(2012, 1, 1))),
```

Comment:
Why did the above need to change?

Reply:
I think how it was before was outdated. It expected a datetime back before, but now we get a Timestamp returned to us every time. And this Timestamp has the functionality of both a Timestamp and a datetime.

Reply:
Keeping it as `datetime` here helps test that `IonEvent`s and `IonPyTimestamp`s work with both `Timestamp` and `datetime`. I would keep it as a `datetime` unless that causes the test to fail, in which case we should understand why.

Both `IonEvent`s and `IonPyTimestamp`s may be created by the user and passed either to a writer coroutine or `simpleion.dump(s)`, respectively. The user may have a `datetime` object, and that should be accepted by ion-python in all the same places as `Timestamp`.

Reply:
Upon investigation, this test expects a datetime back but instead we are getting a `Timestamp` back. We are getting a Timestamp back because in the Timestamp constructor, since `microsecond` is defined and `fractional_seconds` is None, we create a new Decimal from the value of `microsecond`, as detailed here: https://github.com/amzn/ion-python/pull/100#discussion_r325348662
What do you think is the right course of action in this case?

Reply:
Fixed, the issue was found to be in core.py's `IonEvent`'s `__eq__` method. Changed back to the original `datetime`.